v2/spanner-to-sourcedb/src/test/java/com/google/cloud/teleport/v2/templates/dbutils/dml/CassandraTypeHandlerTest.java

```diff
@@ -0,0 +1,478 @@
+        Set<ByteBuffer> value = handleByteSetType("data", newValuesJson);
+
+        Set<ByteBuffer> expected = new HashSet<>(Arrays.asList(
+                ByteBuffer.wrap(Base64.getDecoder().decode("QUJDRA==")),
+                ByteBuffer.wrap(Base64.getDecoder().decode("RkZIRg=="))
+        ));
+        assertEquals(expected, value);
+    }
+
+    @Test
+    public void testHandleStringArrayType() {
+        String newValuesString = "{\"names\":[\"Alice\", \"Bob\", \"Charlie\"]}";
+        JSONObject newValuesJson = new JSONObject(newValuesString);
+        Set<String> value = handleStringArrayType("names", newValuesJson);
+
+        Set<String> expected = new HashSet<>(Arrays.asList("Alice", "Bob", "Charlie"));
+        assertEquals(expected, value);
+    }
+
+    @Test
+    public void testHandleStringSetType() {
+        String newValuesString = "{\"names\":[\"Alice\", \"Bob\", \"Alice\", \"Charlie\"]}";
+        JSONObject newValuesJson = new JSONObject(newValuesString);
+        List<String> valueList = handleStringSetType("names", newValuesJson);
+        Set<String> value = new HashSet<>(valueList);
+        Set<String> expected = new HashSet<>(Arrays.asList("Alice", "Bob", "Charlie"));
+        assertEquals(expected, value);
+    }
+
+    @Test
+    public void testHandleBoolSetTypeString() {
+        String newValuesString = "{\"flags\":[\"true\", \"false\", \"true\"]}";
+        JSONObject newValuesJson = new JSONObject(newValuesString);
+        Set<Boolean> value = handleBoolSetTypeString("flags", newValuesJson);
+
+        Set<Boolean> expected = new HashSet<>(Arrays.asList(true, false));
+        assertEquals(expected, value);
+    }
+
+    @Test
+    public void testHandleFloatArrayType() {
+        String newValuesString = "{\"values\":[1.1, 2.2, 3.3]}";
+        JSONObject newValuesJson = new JSONObject(newValuesString);
+        List<Float> value = handleFloatArrayType("values", newValuesJson);
+
+        List<Float> expected = Arrays.asList(1.1f, 2.2f, 3.3f);
+        assertEquals(expected, value);
+    }
+
+    @Test
+    public void testHandleFloatSetType() {
+        String newValuesString = "{\"values\":[1.1, 2.2, 3.3, 2.2]}";
+        JSONObject newValuesJson = new JSONObject(newValuesString);
+        Set<Float> value = handleFloatSetType("values", newValuesJson);
+
+        Set<Float> expected = new HashSet<>(Arrays.asList(1.1f, 2.2f, 3.3f));
+        assertEquals(expected, value);
+    }
+
+    @Test
+    public void testHandleFloatSetType_InvalidString() {
+        String newValuesString = "{\"values\":[\"1.1\", \"2.2\", \"abc\"]}";
+        JSONObject newValuesJson = new JSONObject(newValuesString);
+        try {
+            handleFloatSetType("values", newValuesJson);
+            fail("Expected IllegalArgumentException for invalid number format");
+        } catch (IllegalArgumentException e) {
+            assertTrue(e.getMessage().contains("Invalid number format for column values"));
```

Comment:
Instead of wrapping these in try catch block , use something like:
```
@Rule public ExpectedException expectedEx = ExpectedException.none();
expectedEx.expect(IllegalArgumentException.class);
expectedEx.expectMessage("Invalid number format for column values");
``` 
Same comment for all methods throwing exception

Reply:
Address the same @shreyakhajanchi